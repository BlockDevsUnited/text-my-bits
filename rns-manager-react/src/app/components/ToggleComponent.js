/* eslint-disable react/self-closing-comp */
import React from 'react';
import propTypes from 'prop-types';

const ToggleComponent = (props) => {
  const {
    labelLeft,
    labelRight,
    onChange,
    value,
  } = props;

  const handleChange = (changeTo) => {
    onChange(changeTo);
  };

  return (
    <div className="toggleSwitch">
      <button
        className={!value ? 'left active' : 'left'}
        onClick={() => handleChange(false)}
        type="button"
      >
        {labelLeft}
      </button>
      <label className="switch" htmlFor="toggleSwitch">
        <input
          type="checkbox"
          onChange={() => handleChange(!value)}
          id="toggleSwitch"
          checked={value}
        />
        <span className="slider"></span>
      </label>
      <button
        className={value ? 'right active' : 'right'}
        onClick={() => handleChange(true)}
        type="button"
      >
        {labelRight}
      </button>
    </div>
  );
};

ToggleComponent.defaultProps = {
  labelLeft: 'leftLabel',
  labelRight: 'rightLabel',
  value: false,
};

ToggleComponent.propTypes = {
  labelLeft: propTypes.string,
  labelRight: propTypes.string,
  onChange: propTypes.func.isRequired,
  value: propTypes.bool,
};

export default ToggleComponent;
