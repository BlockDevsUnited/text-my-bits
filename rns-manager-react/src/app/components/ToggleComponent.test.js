import React from 'react';
import { mount } from 'enzyme';
import ToggleComponent from './ToggleComponent';


describe('toggleComponent', () => {
  const handleChange = jest.fn();
  it('defaults matches snapshot', () => {
    const component = mount(<ToggleComponent onChange={handleChange} />);
    expect(component.find('input[type="checkbox"]').props().value).toBeFalsy();

    const activeButton = component.find('button.active');
    expect(activeButton.text()).toEqual('leftLabel');

    expect(component).toMatchSnapshot();
  });

  it('toggleSwitch active matches snapshot', () => {
    const component = mount(<ToggleComponent onChange={handleChange} value />);
    expect(component.find('input[type="checkbox"]').props().value).toBeFalsy();

    const activeButton = component.find('button.active');
    expect(activeButton.text()).toEqual('rightLabel');

    expect(component).toMatchSnapshot();
  });

  it('sets the text from the parent', () => {
    const component = mount(
      <ToggleComponent labelLeft="LEFT" labelRight="RIGHT" onChange={handleChange} />,
    );

    expect(component.find('button.left').text()).toEqual('LEFT');
    expect(component.find('button.right').text()).toEqual('RIGHT');
  });

  it('calls the function when switched or clicked', () => {
    const component = mount(
      <ToggleComponent labelLeft="LEFT" labelRight="RIGHT" onChange={handleChange} />,
    );

    expect(handleChange).toBeCalledTimes(0);
    component.find('#toggleSwitch').simulate('change', { target: { checked: true } });
    expect(handleChange).toBeCalledTimes(1);

    component.find('button.left').simulate('click');
    expect(handleChange).toBeCalledTimes(2);

    component.find('button.right').simulate('click');
    expect(handleChange).toBeCalledTimes(3);
  });

  it('returns correct value when buttons are clicked', () => {
    const component = mount(
      <ToggleComponent onChange={handleChange} />,
    );

    component.find('button.right').simulate('click');
    expect(handleChange).toBeCalledWith(true);

    component.find('button.left').simulate('click');
    expect(handleChange).toBeCalledWith(false);
  });
});
