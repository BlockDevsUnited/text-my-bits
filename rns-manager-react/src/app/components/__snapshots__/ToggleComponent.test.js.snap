// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toggleComponent defaults matches snapshot 1`] = `
<ToggleComponent
  labelLeft="leftLabel"
  labelRight="rightLabel"
  onChange={[MockFunction]}
  value={false}
>
  <div
    className="toggleSwitch"
  >
    <button
      className="left active"
      onClick={[Function]}
      type="button"
    >
      leftLabel
    </button>
    <label
      className="switch"
      htmlFor="toggleSwitch"
    >
      <input
        checked={false}
        id="toggleSwitch"
        onChange={[Function]}
        type="checkbox"
      />
      <span
        className="slider"
      />
    </label>
    <button
      className="right"
      onClick={[Function]}
      type="button"
    >
      rightLabel
    </button>
  </div>
</ToggleComponent>
`;

exports[`toggleComponent toggleSwitch active matches snapshot 1`] = `
<ToggleComponent
  labelLeft="leftLabel"
  labelRight="rightLabel"
  onChange={[MockFunction]}
  value={true}
>
  <div
    className="toggleSwitch"
  >
    <button
      className="left"
      onClick={[Function]}
      type="button"
    >
      leftLabel
    </button>
    <label
      className="switch"
      htmlFor="toggleSwitch"
    >
      <input
        checked={true}
        id="toggleSwitch"
        onChange={[Function]}
        type="checkbox"
      />
      <span
        className="slider"
      />
    </label>
    <button
      className="right active"
      onClick={[Function]}
      type="button"
    >
      rightLabel
    </button>
  </div>
</ToggleComponent>
`;
